tests/unittests/tests-mtd/tests-mtd.c

```diff
@@ -0,0 +1,260 @@
+    .power = power,
+};
+
+static mtd_dev_t _dev = {
+    .driver = &driver,
+    .sector_count = SECTOR_COUNT,
+    .pages_per_sector = PAGE_PER_SECTOR,
+    .page_size = PAGE_SIZE,
+};
+#endif /* MTD_0 */
+
+static void setup_teardown(void)
+{
+    mtd_dev_t *dev = (mtd_dev_t*) &_dev;
+
+    mtd_erase(dev, 0, dev->pages_per_sector * dev->page_size);
+}
+
+static void test_mtd_init(void)
+{
+    mtd_dev_t *dev = (mtd_dev_t*) &_dev;
+
+    int ret = mtd_init(dev);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+}
+
+static void test_mtd_erase(void)
+{
+    mtd_dev_t *dev = (mtd_dev_t*) &_dev;
+
+    /* Erase first sector */
+    int ret = mtd_erase(dev, 0, dev->pages_per_sector * dev->page_size);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+
+    /* Erase with wrong size (les than sector size) */
+    ret = mtd_erase(dev, 0, dev->page_size);
+    TEST_ASSERT_EQUAL_INT(-EOVERFLOW, ret);
+
+    /* Unaligned erase */
+    ret = mtd_erase(dev, dev->page_size, dev->page_size);
+    TEST_ASSERT_EQUAL_INT(-EOVERFLOW, ret);
+
+    /* Erase 2nd - 3rd sector */
+    ret = mtd_erase(dev, dev->pages_per_sector * dev->page_size,
+                    dev->pages_per_sector * dev->page_size * 2);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+
+    /* Erase out of memory area */
+    ret = mtd_erase(dev, dev->pages_per_sector * dev->page_size * dev->sector_count,
+                    dev->pages_per_sector * dev->page_size);
+    TEST_ASSERT_EQUAL_INT(-EOVERFLOW, ret);
+}
+
+static void test_mtd_write_erase(void)
+{
+    mtd_dev_t *dev = (mtd_dev_t*) &_dev;
+    const char buf[] = "ABCDEFGHIJK";
+    uint8_t buf_empty[] = {0xff, 0xff, 0xff};
+    char buf_read[sizeof(buf) + sizeof(buf_empty)];
+    memset(buf_read, 0, sizeof(buf_read));
+
+    int ret = mtd_write(dev, buf, sizeof(buf_empty), sizeof(buf));
+    TEST_ASSERT_EQUAL_INT(sizeof(buf), ret);
+
+    ret = mtd_erase(dev, 0, dev->pages_per_sector * dev->page_size);
+    TEST_ASSERT_EQUAL_INT(0, ret);
+
+    uint8_t expected[sizeof(buf_read)];
+    memset(expected, 0xff, sizeof(expected));
+    ret = mtd_read(dev, buf_read, 0, sizeof(buf_read));
+    TEST_ASSERT_EQUAL_INT(sizeof(buf_read), ret);
+    TEST_ASSERT_EQUAL_INT(0, memcmp(expected, buf_read, sizeof(buf_read)));
+
+}
+
+static void test_mtd_write_read(void)
+{
+    mtd_dev_t *dev = (mtd_dev_t*) &_dev;
+    const char buf[] = "ABCDEFGH";
+    uint8_t buf_empty[] = {0xff, 0xff, 0xff};
+    char buf_read[sizeof(buf) + sizeof(buf_empty)];
+    memset(buf_read, 0, sizeof(buf_read));
+
+    /* Basic write / read */
+    int ret = mtd_write(dev, buf, 0, sizeof(buf));
+    TEST_ASSERT_EQUAL_INT(sizeof(buf), ret);
+
+    ret = mtd_read(dev, buf_read, 0, sizeof(buf_read));
+    TEST_ASSERT_EQUAL_INT(sizeof(buf_read), ret);
+    TEST_ASSERT_EQUAL_INT(0, memcmp(buf, buf_read, sizeof(buf)));
+    TEST_ASSERT_EQUAL_INT(0, memcmp(buf_empty, buf_read + sizeof(buf), sizeof(buf_empty)));
+
+    /* Unaligned write / read */
+    ret = mtd_write(dev, buf, dev->page_size, sizeof(buf));
```

Comment:
How about we change this to 
```c
ret = mtd_write(dev, buf, dev->page_size + sizeof(buf_empty), sizeof(buf));
```
and change the readback to match:
```c
TEST_ASSERT_EQUAL_INT(0, memcmp(buf_empty, buf_read, sizeof(buf_empty)));
TEST_ASSERT_EQUAL_INT(0, memcmp(buf, buf_read + sizeof(buf_empty), sizeof(buf)));
```
that will test completely unaligned writes, and also catch implementation errors where the read or write addresses are not the same